// react-wood-duck/dist/test/CommonAddressComponent_tests.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['../CommonAddressComponent.js', 'react-dom/lib/ReactTestUtils', 'react', 'enzyme', './Enzyme_setup'], factory);
  } else if (typeof exports !== "undefined") {
    factory(require('../CommonAddressComponent.js'), require('react-dom/lib/ReactTestUtils'), require('react'), require('enzyme'), require('./Enzyme_setup'));
  } else {
    var mod = {
      exports: {}
    };
    factory(global.CommonAddressComponent, global.ReactTestUtils, global.react, global.enzyme, global.Enzyme_setup);
    global.CommonAddressComponent_tests = mod.exports;
  }
})(this, function (_CommonAddressComponent, _ReactTestUtils, _react, _enzyme) {
  'use strict';

  var _CommonAddressComponent2 = _interopRequireDefault(_CommonAddressComponent);

  var _ReactTestUtils2 = _interopRequireDefault(_ReactTestUtils);

  var _react2 = _interopRequireDefault(_react);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  var stateTypes = {
    items: [{
      id: 1,
      value: 'Alabama'
    }, {
      id: 2,
      value: 'Alaska'
    }, {
      id: 3,
      value: 'American Samoa'
    }, {
      id: 4,
      value: 'Arizona'
    }, {
      id: 5,
      value: 'Arkansas'
    }, {
      id: 6,
      value: 'California'
    }]
  };
  describe('Verify Common Address Fields Component', function () {
    var commonAddressComponent = void 0,
        onChangeSpy = void 0;
    onChangeSpy = jasmine.createSpy('onChange');
    var addressFields = {
      street_address: '',
      zip: '',
      city: '',
      state: null
    };
    var props = {
      addressTitle: 'Physical Address',
      addressType: 'Residential',
      id: 'street_address',
      suggestions: [],
      stateTypes: stateTypes.items,
      addressFields: addressFields,
      onChange: onChangeSpy
    };
    beforeEach(function () {
      commonAddressComponent = (0, _enzyme.shallow)(_react2.default.createElement(_CommonAddressComponent2.default, {
        addressTitle: 'Physical Address',
        addressType: 'Residential',
        id: 'street_address',
        suggestions: [],
        stateTypes: stateTypes.items,
        addressFields: addressFields,
        onChange: onChangeSpy,
        renderSuggestion: onChangeSpy,
        onSuggestionsFetchRequested: onChangeSpy
      }));
    });
    it('Load Address Fields', function () {
      expect(commonAddressComponent.length).toEqual(1);
    });
    it('verify Residential street address', function () {
      var streetAddress = commonAddressComponent.find('#Residentialstreet_address').at(0);
      expect(streetAddress.length).toBe(1);
    });
    it('verify mailing address zip change', function () {
      var zipCodeField = commonAddressComponent.find('#Residentialzip').at(0);
      zipCodeField.simulate('change', { target: { value: '95833' } });
      expect(onChangeSpy).toHaveBeenCalledWith('zip', '95833');
    });
    it('verify mailing Address city change', function () {
      var cityField = commonAddressComponent.find('#Residentialcity').at(0);
      cityField.simulate('change', { target: { value: 'Sacramento' } });
      expect(onChangeSpy).toHaveBeenCalledWith('city', 'Sacramento');
    });
    it('verify mailing Address State change', function () {
      var stateDropDownField = commonAddressComponent.find('#Residentialstate_type').at(0);
      stateDropDownField.simulate('change', {
        target: { selectedOptions: [{ value: 1, text: 'Alabama' }] }
      });
      expect(onChangeSpy).toHaveBeenCalledWith('state', {
        id: 1,
        value: 'Alabama'
      });
    });
    describe('AutoSuggest Render', function () {
      var renderedComponent = void 0,
          resultTag = void 0;
      beforeEach(function () {
        renderedComponent = _ReactTestUtils2.default.createRenderer();
        renderedComponent.render(_react2.default.createElement(_CommonAddressComponent2.default, props));
        resultTag = renderedComponent.getRenderOutput();
      });
      it('AutoSuggest label', function () {
        var autoSuggestField = resultTag.props.children[0].props.children[0].props;
        expect(autoSuggestField.children).toBe('Physical Address');
      });
      it('for AutoSuggest Component load', function () {
        var streesAddressAutoFill = resultTag.props.children[0].props.children[1].type.name;
        expect(streesAddressAutoFill).toBe('Autosuggest');
      });
    });
  });
});
