// react-wood-duck/dist/test/NavLinksContainer_test.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['react', 'react-dom/lib/ReactTestUtils', '../NavLinksContainer', '../NavLinks'], factory);
  } else if (typeof exports !== "undefined") {
    factory(require('react'), require('react-dom/lib/ReactTestUtils'), require('../NavLinksContainer'), require('../NavLinks'));
  } else {
    var mod = {
      exports: {}
    };
    factory(global.react, global.ReactTestUtils, global.NavLinksContainer, global.NavLinks);
    global.NavLinksContainer_test = mod.exports;
  }
})(this, function (_react, _ReactTestUtils, _NavLinksContainer, _NavLinks) {
  'use strict';

  var _react2 = _interopRequireDefault(_react);

  var _ReactTestUtils2 = _interopRequireDefault(_ReactTestUtils);

  var _NavLinksContainer2 = _interopRequireDefault(_NavLinksContainer);

  var _NavLinks2 = _interopRequireDefault(_NavLinks);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  describe('NavLinksContainer', function () {
    describe('given empty navigation links', function () {
      var emptyNavLinks = [];
      var renderer = _ReactTestUtils2.default.createRenderer();
      renderer.render(_react2.default.createElement(_NavLinksContainer2.default, { navLinks: emptyNavLinks }));
      var resultTag = renderer.getRenderOutput();

      it('renders NavLinks Component', function () {
        expect(resultTag.type).toBe(_NavLinks2.default);
      });

      it('renders NavLinks component with empty navLinks property', function () {
        expect(resultTag.props.navLinks).toEqual(emptyNavLinks);
      });

      it('renders NavLinks component with null activeNavLinkHref', function () {
        expect(resultTag.props.activeNavLinkHref).toBe(null);
      });
    });

    describe('given navigation links', function () {
      var renderer = _ReactTestUtils2.default.createRenderer();
      var simpleNavLinks = [{
        type: 'navLink',
        text: 'Tommy Cambell',
        href: '#tommy',
        preIcon: 'fa fa-user'
      }, {
        type: 'navLink',
        text: 'Aubrey Cambell',
        href: '#aub',
        preIcon: 'fa fa-user'
      }, {
        type: 'navLink',
        text: 'Chris Cambell',
        href: '#chris',
        preIcon: 'fa fa-user'
      }];
      renderer.render(_react2.default.createElement(_NavLinksContainer2.default, { navLinks: simpleNavLinks }));
      var resultTag = renderer.getRenderOutput();

      it('renders NavLinks component with give navLinks property', function () {
        expect(resultTag.type).toBe(_NavLinks2.default);
        expect(resultTag.props.navLinks).toEqual(simpleNavLinks);
      });

      it('renders NavLinks component with first NavLink as active NavLink', function () {
        expect(resultTag.props.activeNavLinkHref).toBe(simpleNavLinks[0].href);
      });

      it('renders NavLinks component with a nav link click handler', function () {
        expect(resultTag.props.handleClick).toBeTruthy();
      });
    });
  });
});
