// react-wood-duck/dist/DropDownField.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports', 'react', 'prop-types'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require('react'), require('prop-types'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.react, global.propTypes);
    global.DropDownField = mod.exports;
  }
})(this, function (exports, _react, _propTypes) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  var DropDownField = function DropDownField(props) {
    return _react2.default.createElement(
      'div',
      { className: 'form-group' },
      _react2.default.createElement(
        'div',
        { className: props.gridClassName },
        _react2.default.createElement(
          'label',
          null,
          props.label
        ),
        _react2.default.createElement(
          'select',
          {
            name: props.name,
            id: props.id,
            value: props.selectedOption,
            onChange: props.onChange,
            onBlur: props.onChange,
            className: props.selectClassName,
            disabled: props.disable
          },
          _react2.default.createElement(
            'option',
            { value: '' },
            props.placeholder
          ),
          props.options.map(function (opt) {
            return _react2.default.createElement(
              'option',
              { key: opt.id, value: opt.id },
              opt.value
            );
          })
        )
      )
    );
  };

  DropDownField.propTypes = {
    name: _propTypes2.default.string.isRequired,
    id: _propTypes2.default.string,
    options: _propTypes2.default.array.isRequired,
    selectedOption: _propTypes2.default.string,
    onChange: _propTypes2.default.func.isRequired,
    placeholder: _propTypes2.default.string,
    gridClassName: _propTypes2.default.string,
    label: _propTypes2.default.string,
    selectClassName: _propTypes2.default.string,
    disable: _propTypes2.default.bool
  };

  exports.default = DropDownField;
});
