// react-wood-duck/dist/NavLinksContainer.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports', 'react', 'prop-types', './NavLinks'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require('react'), require('prop-types'), require('./NavLinks'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.react, global.propTypes, global.NavLinks);
    global.NavLinksContainer = mod.exports;
  }
})(this, function (exports, _react, _propTypes, _NavLinks) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var _NavLinks2 = _interopRequireDefault(_NavLinks);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _classCallCheck(instance, Constructor) {
    if (!(instance instanceof Constructor)) {
      throw new TypeError("Cannot call a class as a function");
    }
  }

  var _createClass = function () {
    function defineProperties(target, props) {
      for (var i = 0; i < props.length; i++) {
        var descriptor = props[i];
        descriptor.enumerable = descriptor.enumerable || false;
        descriptor.configurable = true;
        if ("value" in descriptor) descriptor.writable = true;
        Object.defineProperty(target, descriptor.key, descriptor);
      }
    }

    return function (Constructor, protoProps, staticProps) {
      if (protoProps) defineProperties(Constructor.prototype, protoProps);
      if (staticProps) defineProperties(Constructor, staticProps);
      return Constructor;
    };
  }();

  function _possibleConstructorReturn(self, call) {
    if (!self) {
      throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
    }

    return call && (typeof call === "object" || typeof call === "function") ? call : self;
  }

  function _inherits(subClass, superClass) {
    if (typeof superClass !== "function" && superClass !== null) {
      throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
    }

    subClass.prototype = Object.create(superClass && superClass.prototype, {
      constructor: {
        value: subClass,
        enumerable: false,
        writable: true,
        configurable: true
      }
    });
    if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
  }

  var NavLinksContainer = function (_React$Component) {
    _inherits(NavLinksContainer, _React$Component);

    function NavLinksContainer(props) {
      _classCallCheck(this, NavLinksContainer);

      var _this = _possibleConstructorReturn(this, (NavLinksContainer.__proto__ || Object.getPrototypeOf(NavLinksContainer)).call(this, props));

      _this.state = { activeNavLinkHref: _this.getDefaultActiveNavLinkHref() };
      _this.handleNavLinkClick = _this.handleNavLinkClick.bind(_this);
      return _this;
    }

    _createClass(NavLinksContainer, [{
      key: 'getDefaultActiveNavLinkHref',
      value: function getDefaultActiveNavLinkHref() {
        return this.props.navLinks && this.props.navLinks.length > 0 ? this.props.navLinks[0].href : null;
      }
    }, {
      key: 'handleNavLinkClick',
      value: function handleNavLinkClick(href, e) {
        this.setState({ activeNavLinkHref: href });
      }
    }, {
      key: 'render',
      value: function render() {
        var navLinks = this.props.navLinks;

        return _react2.default.createElement(_NavLinks2.default, {
          navLinks: navLinks,
          handleClick: this.handleNavLinkClick,
          activeNavLinkHref: this.state.activeNavLinkHref
        });
      }
    }]);

    return NavLinksContainer;
  }(_react2.default.Component);

  NavLinksContainer.propTypes = {
    navLinks: _propTypes2.default.arrayOf(_propTypes2.default.object).isRequired
  };

  exports.default = NavLinksContainer;
});
